CHANGELOG.md

```diff
@@ -1,3 +1,21 @@
+## 2.12.0-beta.1
+
+## Features ✨ and improvements 🏁
+
+* Improve performance of patterns and line dashes, also improving their appearance when zooming ([#12326](https://github.com/mapbox/mapbox-gl-js/pull/12326))
+* Improve performance of text and icon placement. ([#12351](https://github.com/mapbox/mapbox-gl-js/pull/12351))
+* Improve performance of loading terrain data ([#12397](https://github.com/mapbox/mapbox-gl-js/pull/12397))
+* Improve poles support in `getBounds` when projection is set to globe ([#12315](https://github.com/mapbox/mapbox-gl-js/pull/12315))
+* Allow zooming towards terrain at a safe distance without pitching the camera ([#12354](https://github.com/mapbox/mapbox-gl-js/pull/12354))
```

Comment:
Should this be a bugfix? Maybe something like:
```
Fix pitch changing to 0 on zooming in to terrain
```

Reply:
I was also considering moving it to a bug fix but it seems more fitting under the improvement section as the "right" behavior can be quite subjective.

Reply:
Good point! My thinking was that it fixed [this bug](https://github.com/mapbox/mapbox-gl-js/issues/10197) but it looks like the scope of that ticket is smaller. Thoughts on rephrasing to:

```suggestion
* Update collisions with terrain to longer adjust pitch and reduce clipping. Mouse panning into terrain now lifts the camera above terrain. ([#12354](https://github.com/mapbox/mapbox-gl-js/pull/12354))
```

cc @avpeery

Reply:
I personally considered it a bug but seems like both phrasings and categories fit, so whatever works!